lib/saluki-components/src/sources/checks/python_exposed_modules.rs

```diff
@@ -0,0 +1,160 @@
+    println!(
+        "submit_service_check called with name: {}, status: {}, tags: {:?}, hostname: {}, message: {:?}",
+        name, status, tags, hostname, message
+    );
+}
+
+#[pyfunction]
+fn metrics(name: String) -> Vec<String> {
+    println!("metrics called for: {}", name);
+    vec![] // Dummy return
+}
+
+#[pyfunction]
+fn reset() {
+    println!("reset called");
+}
+
+#[pymodule]
+pub fn aggregator(m: &Bound<'_, PyModule>) -> PyResult<()> {
+    m.add_function(wrap_pyfunction_bound!(submit_metric, m)?)?;
+    m.add_function(wrap_pyfunction_bound!(submit_service_check, m)?)?;
+    m.add_function(wrap_pyfunction_bound!(self::metrics, m)?)?;
+    m.add_function(wrap_pyfunction_bound!(reset, m)?)?;
+
+    m.add("GAUGE", PyMetricType::Gauge as i32)?;
+    m.add("RATE", PyMetricType::Rate as i32)?;
+    m.add("COUNT", PyMetricType::Count as i32)?;
+    m.add("MONOTONIC_COUNT", PyMetricType::MonotonicCount as i32)?;
+    m.add("COUNTER", PyMetricType::Counter as i32)?;
+    m.add("HISTOGRAM", PyMetricType::Histogram as i32)?;
+    m.add("HISTORATE", PyMetricType::Historate as i32)?;
+
+    Ok(())
+}
+
+#[pyfunction]
+fn get_hostname() -> &'static str {
+    trace!("Called get_hostname()");
+    "stubbed.hostname"
+}
+
+#[pyfunction]
+fn set_hostname(hostname: String) {
+    trace!("Called set_hostname({})", hostname);
+    // In a function context without a struct, we cannot actually "set" the hostname persistently.
+}
+
+#[pyfunction]
+fn reset_hostname() {
+    trace!("Called reset_hostname()");
+    // Similar to `set_hostname`, we cannot reset without a persistent structure.
+}
+
+#[pyfunction]
+fn get_config(config_option: String) -> bool {
+    trace!("Called get_config({})", config_option);
+
+    false
+}
+
+#[pyfunction]
+fn get_version() -> &'static str {
+    trace!("Called get_version()");
+    "0.0.0"
+}
+
+#[pyfunction]
+fn log(message: String, level: u32) {
+    match level {
+        // All except trace are from python3 logging levels
+        // https://docs.python.org/3/library/logging.html#levels
+        // Trace is manually specified in datadog_checks_base
+        // https://github.com/DataDog/integrations-core/blob/458274dfd867b40e368c795574b6d97a9b7e471d/datadog_checks_base/datadog_checks/base/log.py#L20-L21
+        40 => tracing::event!(tracing::Level::ERROR, "Python Log: {}", message),
+        30 => tracing::event!(tracing::Level::WARN, "Python Log: {}", message),
+        20 => tracing::event!(tracing::Level::INFO, "Python Log: {}", message),
+        10 => tracing::event!(tracing::Level::DEBUG, "Python Log: {}", message),
+        7 => tracing::event!(tracing::Level::TRACE, "Python Log: {}", message),
+        _ => tracing::event!(tracing::Level::TRACE, "Python Log: {}", message),
+    };
+}
+
+#[pyfunction]
+fn set_check_metadata(check_id: String, name: String, value: String) {
+    debug!("Called set_check_metadata({}, {}, {})", check_id, name, value);
```

Comment:
I never dug into this to see what this check_metadata is actually used for/by, this could be a tricky one with race conditions around checks being scheduled/descheduled.